admin/class-taxonomy-presenter.php

```diff
@@ -0,0 +1,317 @@
+						'type'        => 'text',
+						'options'     => array( 'class' => 'large-text' ),
+					),
+					$settings['network'] . '-description' => array(
+						'label'       => sprintf( __( '%s Description', 'wordpress-seo' ), $settings['label'] ),
+						'description' => sprintf( esc_html__( 'If you don\'t want to use the meta description for sharing on %1$s but want another description there, write it here.', 'wordpress-seo' ), $settings['label'] ),
+						'type'        => 'textarea',
+						'options'     => '',
+
+					),
+					$settings['network'] . '-image' => array(
+						'label'       => sprintf( __( '%s Image', 'wordpress-seo' ), $settings['label'] ),
+						'description' => sprintf( esc_html__( 'If you want to use an image for sharing on %1$s, you can upload / choose an image or add the image URL here.', 'wordpress-seo' ), $settings['label'] ) . '<br />' .
+							sprintf( __( 'The recommended image size for %1$s is %2$spx.', 'wordpress-seo' ), $settings['label'], $settings['size'] ),
+						'type'        => 'upload',
+						'options'     => '',
+
+					),
+				);
+
+				$fields = array_merge( $fields, $fields_to_push );
+
+			}
+		}
+
+		return $fields;
+	}
+
+	/**
+	 * Returns a bool to determine if the social tab has to be visible
+	 *
+	 * @return bool
+	 */
+	public function show_social() {
+		return ( $this->options['opengraph'] === true || $this->options['twitter'] === true || $this->options['googleplus'] === true );
+	}
+
+	/**
+	 * Translate options text strings for use in the select fields
+	 *
+	 * @internal IMPORTANT: if you want to add a new string (option) somewhere, make sure you add
+	 * that array key to the main options definition array in the class WPSEO_Taxonomy_Meta() as well!!!!
+	 */
+	private function translate_meta_options() {
+		$this->no_index_options        = WPSEO_Taxonomy_Meta::$no_index_options;
+		$this->sitemap_include_options = WPSEO_Taxonomy_Meta::$sitemap_include_options;
+
+		$this->no_index_options['default'] = __( 'Use %s default (Currently: %s)', 'wordpress-seo' );
+		$this->no_index_options['index']   = __( 'Always index', 'wordpress-seo' );
+		$this->no_index_options['noindex'] = __( 'Always noindex', 'wordpress-seo' );
+
+		$this->sitemap_include_options['-']      = __( 'Auto detect', 'wordpress-seo' );
+		$this->sitemap_include_options['always'] = __( 'Always include', 'wordpress-seo' );
+		$this->sitemap_include_options['never']  = __( 'Never include', 'wordpress-seo' );
+	}
+
+	/**
+	 * Create a row in the form table.
+	 *
+	 * @param string $field_name      Variable the row controls.
+	 * @param string $label    Label for the variable.
+	 * @param string $description     Description of the use of the variable.
+	 * @param string $type     Type of form row to create.
+	 * @param array  $options  Options to use when form row is a select box.
+	 */
+	private function form_row( $field_name, $label, $description, $type = 'text', $options = array() ) {
```

Comment:
This function has a lot of responsibilities. Perhaps it's wise to split this up and even move this to a new class.